test/golden/artman_golden_test.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Artman golden tests.
+
+It runs GAPIC generation against the golden input data, can then compare the
+output layout with the expected output. This helps to capture unexpected GAPIC
+output layout change caused by either by artman, toolkit or other runtime
+dependency.
+"""
+
+import os
+import subprocess
+
+import pytest
+
+def test_library_example(clone_googleapis_for_testing=False):
+    golden_dir = os.path.dirname(os.path.realpath(__file__))
+    if clone_googleapis_for_testing:
+        subprocess.check_call([
+            'git',
+            'clone',
+            'https://github.com/googleapis/googleapis',
+            os.path.join(golden_dir, 'googleapis')
+        ])
+    else:
+        pytest.skip(
+            'Skip the golden test as the --clone_googleapis_for_testing is not '
+            'passed through py.test')
+
+    with open(os.path.join(golden_dir, 'library_example.golden')) as f:
+        expected = set(f.read().splitlines())
```

Comment:
Nitpick: Consider changing this to:

```python
expected = set()
for line in f.read().splitlines():
    line = line.strip()
    if line and not line.startswith('#'):
        expected.add(line)
```

This will have the following effects:

1. Blank lines will be skipped (important if a future committer has an editor that automatically adds blank lines to the end of the file, as many do).

2. Any line starting with `#` will be skipped, allowing comments to exist in the baseline file if needed in the future.